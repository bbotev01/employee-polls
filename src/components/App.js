import '../App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import LoginPage from "./LoginPage";
import Dashboard from "./Dashboard";
import Navbar from "./Navbar";
import CreatePoll from './CreatePoll';
import Poll from './Poll';
import Leaderboard from './Leaderboard';
import NotFound from './NotFound';
import { handleInitialData } from "../actions/shared";
import { useEffect, Fragment } from "react";
import LoadingBar from "react-redux-loading-bar";
import { connect } from "react-redux";
import { Routes, Route } from "react-router-dom";

function App(props) {

  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  if (props.loading === true) {
    return (
      <LoadingBar />
    )
  } else if (props.authorized === true) {
    return (
      <Fragment>
        <Navbar />
        <div className="App">
          <Routes>
            <Route exact path="/" element={<Dashboard />} />
            <Route exact path="/leaderboard" element={<Leaderboard />} />
            <Route exact path="/add" element={<CreatePoll />} />
            <Route path="/question/:id" element={<Poll />} />
            <Route path="/login" element={<LoginPage />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </div>
      </Fragment>

    )
  } else {
    return (
      <LoginPage />
    )
  }
}

const mapStateToProps = ({ authedUser, loadingBar }) => ({
  loading: loadingBar.default === 1,
  authorized: authedUser !== null
});

export default connect(mapStateToProps)(App);
