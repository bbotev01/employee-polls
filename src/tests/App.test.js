import { render, screen, fireEvent, } from '@testing-library/react';
import App from '../components/App';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../app/store';

describe('App', () => {

    it('test LoginPage component matches snapshot', async () => {
        var component = render(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <App />
                </Provider>
            </MemoryRouter>)
        expect(await screen.findByTestId('username-input')).toBeInTheDocument();
        expect(await screen.findByTestId('password-input')).toBeInTheDocument();
        expect(await screen.findByTestId('submit')).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });

    it('test App shows LoginPage component and contains user and password inputs and submit button', async () => {
        render(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <App />
                </Provider>
            </MemoryRouter>)
        expect(await screen.findByTestId('username-input')).toBeInTheDocument();
        expect(await screen.findByTestId('password-input')).toBeInTheDocument();
        expect(await screen.findByTestId('submit')).toBeInTheDocument();
    });


    it('test correct App login followed by render of dashboard', async () => {

        var component = render(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <App />
                </Provider>
            </MemoryRouter>)

        expect(await screen.findByTestId('username-input')).toBeInTheDocument();
        expect(await screen.findByTestId('password-input')).toBeInTheDocument();
        expect(await screen.findByTestId('submit')).toBeInTheDocument();
        var username = component.getByTestId('username-input');
        var password = component.getByTestId('password-input');
        var submitButton = component.getByTestId('submit');
        fireEvent.change(username, { target: { value: 'sarahedo' } });
        fireEvent.change(password, { target: { value: 'password123' } });
        fireEvent.click(submitButton);
        expect(await screen.findByTestId('dashboard')).toBeInTheDocument();
    });

    it('test incorrect App login and showing alert', async () => {

        var component = render(
            <MemoryRouter initialEntries={["/"]}>
                <Provider store={store}>
                    <App />
                </Provider>
            </MemoryRouter>)

        expect(await screen.findByTestId('username-input')).toBeInTheDocument();
        expect(await screen.findByTestId('password-input')).toBeInTheDocument();
        expect(await screen.findByTestId('submit')).toBeInTheDocument();
        var username = component.getByTestId('username-input');
        var password = component.getByTestId('password-input');
        var submitButton = component.getByTestId('submit');
        fireEvent.change(username, { target: { value: 'nobody' } });
        fireEvent.change(password, { target: { value: 'password123' } });
        fireEvent.click(submitButton);
        expect(component.getByTestId('alert')).toBeInTheDocument();
    });
});