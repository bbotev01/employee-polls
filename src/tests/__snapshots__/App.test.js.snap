// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test LoginPage component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          Employee Polls
        </h1>
        <svg
          fill="currentColor"
          height="420"
          stroke="currentColor"
          stroke-width="0"
          style="color: gray;"
          viewBox="0 0 448 512"
          width="420"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM160 368c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V240c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v128zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V144c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v224zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-64c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v64z"
          />
        </svg>
        <div
          class="row"
        >
          <div
            class="col-3"
          />
          <div
            class="col-6"
          >
            <form
              class=""
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="username"
                >
                  Username
                </label>
                <input
                  class="form-control"
                  data-testid="username-input"
                  id="username"
                  placeholder="Enter username"
                  type="text"
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="password"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  data-testid="password-input"
                  id="password"
                  placeholder="Enter password"
                  type="password"
                />
              </div>
              <button
                class="btn btn-primary"
                data-testid="submit"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        Employee Polls
      </h1>
      <svg
        fill="currentColor"
        height="420"
        stroke="currentColor"
        stroke-width="0"
        style="color: gray;"
        viewBox="0 0 448 512"
        width="420"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM160 368c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V240c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v128zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V144c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v224zm96 0c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-64c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v64z"
        />
      </svg>
      <div
        class="row"
      >
        <div
          class="col-3"
        />
        <div
          class="col-6"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="username"
              >
                Username
              </label>
              <input
                class="form-control"
                data-testid="username-input"
                id="username"
                placeholder="Enter username"
                type="text"
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                data-testid="password-input"
                id="password"
                placeholder="Enter password"
                type="password"
              />
            </div>
            <button
              class="btn btn-primary"
              data-testid="submit"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
